preprocessing/preprocess.py

```diff
@@ -0,0 +1,157 @@
+"""To run this GOOGLE_APPLICATION_CREDENTIALS should be set.
+
+This script converts the compressed patient info in ELVOs_anon
+into numpy files, saved in the numpy folder of the elvo bucket.
+"""
+import io
+import logging
+import os
+import shutil
+import subprocess
+from typing import List
+
+import numpy as np
+import pandas as pd
+import pydicom
+from google.cloud import storage
+
+import parsers
+import transforms
+
+ELVOS_ANON = 'ELVOs_anon'
+EXTENSION_LENGTH = len('.cab')  # == 4
+
+
+def load_metadata(bucket):
+    """Loads the metadata from GCS.
+    """
+    positives_blob = bucket.get_blob('metadata/positives.csv')
+    positives_bytes = positives_blob.download_as_string()
+    positives_str = positives_bytes.decode('utf-8')
+    positives_df = pd.read_csv(io.StringIO(positives_str))
+
+    negatives_blob = bucket.get_blob('metadata/negatives.csv')
+    negatives_bytes = negatives_blob.download_as_string()
+    negatives_str = negatives_bytes.decode('utf-8')
+    negatives_df = pd.read_csv(io.StringIO(negatives_str))
+    return positives_df, negatives_df
+
+
+def process_cab(blob: storage.Blob, patient_id: str) -> None:
+    os.mkdir('tmp')
+    os.chdir('tmp')
+
+    blob.download_to_filename(patient_id + '.cab')
+    subprocess.call(['cabextract', patient_id + '.cab'])
+    logging.info('extracted cab file')
+
+    dirpath = list(os.walk('.'))[2][0]
+    logging.info(f'loading scans from {dirpath}')
+    processed_scan = _process_cab(dirpath)
+
+    os.chdir('..')
+    shutil.rmtree('tmp')
+
+    save_to_gcs(processed_scan, patient_id, blob.bucket)
+
+
+def _process_cab(dirpath: str) -> np.array:
+    scan = parsers.load_scan(dirpath)
+    processed_scan = preprocess_scan(scan)
+    return processed_scan
+
+
+def process_zip(blob: storage.Blob, patient_id: str) -> None:
+    os.mkdir('tmp')
+    os.chdir('tmp')
+
+    blob.download_to_filename(patient_id + '.zip')
+    logging.info('extracting zip file')
+    shutil.unpack_archive(patient_id + '.zip', format='zip')
+
+    dirpath = list(os.walk('.'))[3][0]
+    logging.info(f'loading scans from {dirpath}')
+    scan = parsers.load_scan(dirpath)
+    processed_scan = preprocess_scan(scan)
```

Comment:
This part looks the same as _process_cab(), easy to refactor

Reply:
I'm going to skip this for now, I did _process_cab() so I could test locally